core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -827,6 +828,33 @@ object TestUtils extends Logging {
     byteBuffer
   }
 
+  /**
+    * Wait until all brokers know about the given list of brokers.
+    *
+    * @param servers The list of servers that the metadata should reach to
+    * @param brokerList The list of brokers
+    * @param timeout The amount of time waiting on this condition before assert to fail
+    * @return The leader of the partition.
+    */
+  def waitUntilBrokerMetadataIsPropagated(servers: Seq[KafkaServer],
+                                          brokerList: Seq[String],
```

Comment:
It's more flexible if we can specify which brokers we want to wait for.  Also, this way is a lot easier since we don't have to pull the host:port pairs out of the KafkaServer objects.